sdk/rust/oak_config_serializer/src/main.rs

```diff
@@ -87,35 +87,55 @@ impl Default for InitialNodeConfig {
     }
 }
 
+/// Directory used to save downloaded Wasm modules.
+/// Created in the `std::env::current_dir()`.
+const CACHE_DIRECTORY: &str = ".oak";
+
 /// Computes SHA256 sum from `data` and returns it as a HEX encoded string.
 fn get_sha256(data: &[u8]) -> String {
     let mut hasher = Sha256::new();
     hasher.update(data);
     hex::encode(hasher.finalize().as_slice().to_vec())
 }
 
-/// Load Wasm module from URL or a file, if URL is not specified.
+/// Download file from `url`.
+async fn download_module_from_url(url: &str) -> anyhow::Result<Vec<u8>> {
+    let url: Url = url.parse().context("Couldn't parse URL")?;
+
+    debug!("Downloading module from: {}", url);
+    let response = reqwest::get(url.clone())
+        .await
+        .with_context(|| format!("Couldn't download module from {}", url))?;
+    let data = response
+        .bytes()
+        .await
+        .context("Couldn't retrieve module from HTTP response")?
+        .to_vec();
+    Ok(data)
+}
+
+/// Load Wasm module from file or URL if specified.
+/// If the file was downloaded from URL, it is cached in [`CACHE_DIRECTORY`].
 async fn load_module(module: &Module) -> anyhow::Result<Vec<u8>> {
-    match module {
+    let data = match &module {
         Module::Path(path) => {
             fs::read(&path).with_context(|| format!("Couldn't read file {}", path))
         }
         Module::External(external) => {
-            let url: Url = external.url.parse().context("Couldn't parse URL")?;
-
-            debug!("Downloading module from: {}", url);
-            // TODO(#1240): Add a Wasm module cache.
-            let response = reqwest::get(url.clone())
-                .await
-                .with_context(|| format!("Couldn't download module from {}", url))?;
-            let data = response
-                .bytes()
-                .await
-                .context("Couldn't retrieve module from HTTP response")?
-                .to_vec();
+            let mut cache_path = std::env::current_dir().unwrap();
+            cache_path.push(CACHE_DIRECTORY);
+            std::fs::create_dir_all(cache_path.as_path())
+                .context("Couldn't create cache directory")?;
 
+            let data = download_module_from_url(&external.url).await?;
```

Comment:
I may be missing something, but it looks to me that this is downloading the file every time, so how does the caching actually work?

Reply:
Thanks!
I somehow lost the most important functionality
Looks like WFH finally started affecting me =)